src/proxy.conf

```diff
@@ -7,6 +7,12 @@
   ProxyRequests Off
   ProxyPreserveHost On
 
+  RewriteEngine on
+  RewriteCond %{HTTP:Upgrade} =websocket
+  RewriteRule /EC2_INSTANCE_ID/(.*)     ws://localhost:8787/$1  [P,L]
+  RewriteCond %{HTTP:Upgrade} !=websocket
+  RewriteRule /EC2_INSTANCE_ID/(.*)     http://localhost:8787/$1 [P,L]
```

Comment:
Do these statements guarantee that all proxied traffic hits the PythonHandler?

Reply:
I verified using burpsuite that it does still go through the PythonHandler.

Here's what I did 
1.modified `access.py` to only do a simple `==` check for the header:
```
  jwt = req.headers_in['x-amzn-oidc-data'] #proxy.conf ensures this header exists

  if jwt == 'foobar':
    return apache.OK
  else:
    return apache.HTTP_UNAUTHORIZED #the userid claim does not match the userid tag
```
2. In intercept: Disable default "File Extension" matcher and add new matcher for URL of type:
```
http://IP_ADDRESS/EC2_INSTANCE_ID/p/*
```
![image](https://user-images.githubusercontent.com/8023327/114230365-eb963580-992d-11eb-9e4e-509ab2c3233c.png)

3. automatically inject `x-amzn-oidc-data: foobar` in "Match and Replace" so that all other requests go through normally
![image](https://user-images.githubusercontent.com/8023327/114230547-1da79780-992e-11eb-87ff-27cd7ab5b4c4.png)

4. Turn on interception in burpsuite

5. Start a new terminal:
![image](https://user-images.githubusercontent.com/8023327/114231107-d8379a00-992e-11eb-9977-7cb6b6869d75.png)


Result:

Note: the `x-amzn-oidc-data` header that burpsuite injected does not get recorded in the HTTP history

If I simply forward `x-amzn-oidc-data: foobar` in the header during interception of `http://IP_ADDRESS/EC2_INSTANCE_ID/p/*` , the websocket is established correctly.
![image](https://user-images.githubusercontent.com/8023327/114231672-965b2380-992f-11eb-8fa4-0f65689c41a6.png)


If I change the `x-amzn-oidc-data` header to any other value, I get a `401` error. 
![image](https://user-images.githubusercontent.com/8023327/114231436-5005c480-992f-11eb-8c42-021aaa942b54.png)